gradle.properties

```diff
@@ -16,8 +16,8 @@
 jmhOutputPath=build/reports/jmh/human-readable-output.txt
 jmhJsonOutputPath=build/reports/jmh/results.json
 jmhIncludeRegex=.*
-systemProp.defaultFlinkVersions=1.20
-systemProp.knownFlinkVersions=1.18,1.19,1.20
+systemProp.defaultFlinkVersions=1.20,2.0
+systemProp.knownFlinkVersions=1.18,1.19,1.20,2.0
```

Comment:
we typically remove the oldest version (1.18) in this case to avoid too many CI builds. that could be a separate commit in this PR. build file change for 1.18 can be combined in this commit.

Reply:
We will use `rebase and merge` for this PR to keep the individual commits. 

Before this is merged, we need to have 4 commits. See examples here
https://github.com/apache/iceberg/pull/10881/commits

Reply:
+1 for removing 1.18. I'll push a separate commit for the removal.

Reply:
I've added the commit and rebased the PR.